src/app/app.module.ts

```diff
@@ -29,7 +30,8 @@ import {UserService} from './user.service'
     AppComponent,
     LoginComponent,
     SelectRepositoryComponent,
-    CohortBuilderComponent
+    CohortBuilderComponent,
+    SignInComponent
```

Comment:
`AppComponent` should of course know about `SignInComponent` because it uses it. It does not makes sense for the root-level module to know about it.

Maybe I'm thinking about this incorrectly. I think of components in a UI as functions in a language. I might create some utility functions that are used in a few places, but the main module knows nothing about them because it describes the high-level functionality of the application.

In this case, I'm *not* telling `AppComponent` about `SignInComponent`, which is exactly what I want to do. Instead, I'm telling the root module about `SignInComponent`, which doesn't seem reasonable.